repositories/AnswerRepository.js

```diff
@@ -8,14 +8,17 @@ const toDb = mapFields(invert(mapping));
 
 module.exports.findAll = function findAll(
   where,
+  whereNot = { state: "Deleted" },
   orderBy = "created_at",
   direction = "asc"
 ) {
-  return Answer.findAll(where).orderBy(orderBy, direction).then(map(fromDb));
+  return Answer.findAll(where, whereNot)
+    .orderBy(orderBy, direction)
+    .then(map(fromDb));
```

Comment:
I've just realised, since the DB layer returns the knex object, we could actually add this functionality without touching that layer at all. This should be identical functionally:

```js
return Answer.findAll(where)
  .whereNot(whereNot)
  .orderBy(orderBy, direction)
  .then(map(fromDb));
```

I think most (all?) of your changes to the access layer can be reverted if we follow this pattern